editor/src/com/talosvfx/talos/editor/nodes/widgets/CustomVarWidget.java

```diff
@@ -0,0 +1,455 @@
+            stageRef = getStage();
+        } else {
+            if (stageRef != null) {
+                stageRef.getRoot().removeCaptureListener(stageListener);
+                stageRef = null;
+            }
+        }
+    }
+
+    private void showEditMode() {
+        if (editing.isVisible()) return;
+
+        if (getStage() == null) {
+            return;
+        }
+        getStage().setKeyboardFocus(fieldNameTextField);
+        fieldNameTextField.selectAll();
+
+        editing.setVisible(true);
+        main.setVisible(false);
+
+        isSelected = true;
+        setBackgrounds();
+    }
+
+    @Override
+    public void act(float delta) {
+        super.act(delta);
+        if (isHover) {
+            CursorUtil.setDynamicModeCursor(CursorUtil.CursorType.MOVE_ALL_DIRECTIONS);
+        }
+    }
+
+    private void hideEditMode() {
+        setFieldName(fieldNameTextField.getText());
+
+        fieldNameTextField.clearSelection();
+
+        editing.setVisible(false);
+        main.setVisible(true);
+
+        isSelected = false;
+        setBackgrounds();
+    }
+
+    private void setBackgrounds() {
+
+        ColorLibrary.BackgroundColor color = ColorLibrary.BackgroundColor.LIGHT_GRAY;
+
+        if (isSelected) {
+            color = ColorLibrary.BackgroundColor.MID_GRAY;
+        } else {
+            if (isHover) {
+                color = ColorLibrary.BackgroundColor.BRIGHT_GRAY;
+            }
+        }
+
+        fieldContainer.setBackground(ColorLibrary.obtainBackground(getSkin(), ColorLibrary.SHAPE_SQUARE, color));
+    }
+
+    public void setFieldName(String text) {
+        String oldName = propertyWrapper.propertyName;
+
+        fieldNameLabel.setText(text);
+        fieldNameTextField.setText(text);
+
+        propertyWrapper.propertyName = text;
+
+        fireNameChangedEvent(oldName, text, false);
+    }
+
+    protected void fireValueChangedEvent(boolean isFastChange) {
+        CustomVarWidgetChangeListener.CustomVarChangeEvent nameChangedEvent = Pools.obtain(CustomVarWidgetChangeListener.CustomVarChangeEvent.class);
+        nameChangedEvent.setType(CustomVarWidgetChangeListener.Type.valueChanged);
+        nameChangedEvent.setFastChange(isFastChange);
+
+        try {
+            fire(nameChangedEvent);
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            Pools.free(nameChangedEvent);
+        }
+    }
+
+    protected void fireNameChangedEvent(String oldName, String newName, boolean isFastChange) {
+        CustomVarWidgetChangeListener.CustomVarChangeEvent nameChangedEvent = Pools.obtain(CustomVarWidgetChangeListener.CustomVarChangeEvent.class);
+        nameChangedEvent.setType(CustomVarWidgetChangeListener.Type.nameChanged);
+        nameChangedEvent.setOldName(oldName);
+        nameChangedEvent.setNewName(newName);
+        nameChangedEvent.setFastChange(isFastChange);
+
+        try {
+            fire(nameChangedEvent);
```

Comment:
Why does firing an event require try catch?

Reply:
copied code, so no clue. Probably the person who wrote it had an issue where its event triggered invoke on another method, which crash and we have this.

Reply:
fixed.